File: app/src/main/java/com/peterlaurence/trekadvisor/util/gpxparser/model/Gpx.java
package com.peterlaurence.trekadvisor.util.gpxparser.model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * GPX documents contain a metadata header, followed by waypoints, routes, and tracks. <p>
 * Custom elements can be added to the extensions section of the GPX document.
 *
 * @author peterLaurence on 12/02/17.
 */
public class Gpx {
    private final List<Track> mTracks;

    private Gpx(Builder builder) {
        mTracks = Collections.unmodifiableList(new ArrayList<>(builder.mTracks));
    }

    public List<Track> getTracks() {
        return mTracks;
    }

    public static class Builder {
        private List<Track> mTracks;

        public Builder setTracks(List<Track> tracks) {
            mTracks = tracks;
            return this;
        }

        public Gpx build() {
            return new Gpx(this);
        }
    }
}

File: app/src/main/java/com/peterlaurence/trekme/util/gpx/model/Gpx.kt
package com.peterlaurence.trekme.util.gpx.model

/**
 * GPX documents have a version and a creator as attributes, and contains an optional metadata header,
 * followed by waypoints, routes, and tracks.
 *
 * Custom elements can be added to the extensions section of the GPX document.
 *
 * @author P.Laurence on 12/02/17.
 */
data class Gpx(
        val metadata: Metadata? = null,
        val tracks: List<Track>,
        val wayPoints: List<TrackPoint>,
        val creator: String = "",
        var version: String = "1.1"
)

/**
 * For instance, only trust [ElevationSource.IGN_RGE_ALTI].
 */
fun Gpx.hasTrustedElevations(): Boolean {
    return metadata?.elevationSourceInfo?.elevationSource == ElevationSource.IGN_RGE_ALTI
}

fun Gpx.getElevationSource(): ElevationSource {
    return metadata?.elevationSourceInfo?.elevationSource ?: ElevationSource.UNKNOWN
}
